File: 100-days-of-code/Beginner/Day05/fizzbuzz.py
def main():
    modulo = {3: "fizz", 5: "buzz"}
    user_input = int(input("What number would you like to fizzbuzz up to? "))
    for i in range(1, user_input + 1):
        output = []
        for num in modulo:
            if i % num == 0:
                output.append(modulo[num])
        if output == []:
            output = [str(i)]
        print("".join(output))


if __name__ == "__main__":
    main()

File: learn-julia/fizzbuzz.jl
function main()
    num_prompt = "Enter a number:"
    number = validate_input(Int, num_prompt)
    println("You entered $number")
    fizzbuzz(number)
end

function validate_input(input_type, prompt)
    res = nothing
    while res == nothing
        print("$prompt ")
        input = readline()
        try
            res = parse(input_type, input)
        catch err
            println("$err")
        end
    end
    return res
end

function fizzbuzz(upper_limit)
    subs = Dict(3 => "fizz", 5 => "buzz")
    for i in 1:upper_limit
        output = ""
        for key in keys(subs)
            if i % key == 0
                output *= subs[key]
            end
        end
        println(isempty(output) ? i : output)
    end
end

if abspath(PROGRAM_FILE) == @__FILE__
    main()
end